### PR TITLE
*(dm): move some SQL utils under BaseDB/BaseConn

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #4287 

### What is changed and how it works?

BaseDB/BaseConn can retry on errors, this PR let some SQL utils use BaseDB/BaseConn

#### Tests 

 - pass existing test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: dm/dm/config/subtask.go [hunk 1]
```diff
@@ -143,6 +146,16 @@ func (db *DBConfig) AdjustWithTimeZone(timeZone string) {
 	db.Adjust()
 }
 
+// FetchTimeZoneSetting fetch target db global time_zone setting.
+// TODO: move this function from TiDB to tiflow.
```
Comment (D3Hunter) [inline dm/dm/config/subtask.go:150]: what does this todo mean?

File: dm/pkg/log/string.go [hunk 2]
```diff
@@ -11,11 +11,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package utils
```
Comment (D3Hunter) [inline dm/pkg/log/string.go:14]: why move this file, static func which doesn't depends on any other package better stay in pkg/utils

File: dm/dm/config/subtask.go (hunk 1 above)
Comment (lance6716) [reply dm/dm/config/subtask.go:150]: I should refer to dbutil.GetTimeZoneOffset and FormatTimeZoneOffset. Will update the comment later

File: dm/pkg/log/string.go (hunk 2 above)
Comment (lance6716) [reply dm/pkg/log/string.go:14]: I remember other files in `utils` will cause a cycle import. Will try to revert it later.

File: dm/pkg/conn/baseconn.go [hunk 3]
```diff
@@ -23,14 +23,14 @@ import (
 	gmysql "github.com/go-mysql-org/go-mysql/mysql"
 	"github.com/go-sql-driver/mysql"
 	"github.com/pingcap/failpoint"
+	"github.com/pingcap/tiflow/dm/pkg/utils"
```
Comment (D3Hunter) [inline dm/pkg/conn/baseconn.go:26]: import groups messed up, other files too

can set per-project import group setting in goland

**Review by D3Hunter — APPROVED** [review]
rest lgtm

**Review by buchuitoudegou — COMMENTED** [review]
rest lgtm. It seems some of the functions in the `utils` pkg (e.g. GetFlavor, GetSlaveServerID) have not been utilizing `baseDB`. Will it be done in future pr or any further consideration around that?

**Comment by lance6716** [comment]
> rest lgtm. It seems some of the functions in the `utils` pkg (e.g. GetFlavor, GetSlaveServerID) have not been utilizing `baseDB`. Will it be done in future pr or any further consideration around that?

I'll open a tracking issue for them, they should also be done.

File: dm/dm/config/subtask.go [hunk 4]
```diff
@@ -143,6 +146,16 @@ func (db *DBConfig) AdjustWithTimeZone(timeZone string) {
 	db.Adjust()
 }
 
+// FetchTimeZoneSetting fetch target db global time_zone setting.
+// TODO: move GetTimeZoneOffset and FormatTimeZoneOffset from TiDB to tiflow.
+func FetchTimeZoneSetting(ctx context.Context, db *sql.DB) (string, error) {
+	dur, err := dbutil.GetTimeZoneOffset(ctx, db)
```
[the remark this answers is not in the document]
Comment (lance6716) [reply dm/dm/config/subtask.go:152]: this PR does not take care about the function

File: dm/pkg/checker/binlog.go [hunk 5]
```diff
@@ -207,12 +211,13 @@ func (c *BinlogDBChecker) Check(ctx context.Context) *Result {
 		Extra: fmt.Sprintf("address of db instance - %s:%d", c.dbinfo.Host, c.dbinfo.Port),
 	}
 
-	flavor, err := utils.GetFlavor(ctx, c.db)
+	flavor, err := utils.GetFlavor(ctx, c.db.DB)
```
[the remark this answers is not in the document]
Comment (lance6716) [reply dm/pkg/checker/binlog.go:214]: not that easy. will done in other PR

**Review by buchuitoudegou — APPROVED** [review]

**Pull request merged**